### PR TITLE
devtoolbox: init at 1.1.1 (and dependencies)

### PR DESCRIPTION
**Pull request by Aleksanaa**
## Description of changes

Tested

image

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Aleksanaa** [comment]
Not working:

- json yaml conversion
- qrcode generation

**Review by ghost — COMMENTED** [review]
can you submit the patch upstream?

there seem to be tests for the python modules, can you add `pytestCheckHook` and run the tests?

File: pkgs/development/python-modules/uuid6/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  buildInputs = [
+    setuptools
+  ];
+
+  pythonImportsCheck = [
+    "uuid6"
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/uuid6/default.nix:26]: Please execute the tests https://github.com/oittaa/uuid6-python/tree/main/test

File: pkgs/development/python-modules/uuid6/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  pname = "uuid6";
+  disabled = pythonOlder "3.8";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-7Qr7OpcwV1dfmIMgG67+QCeHyl4R4dJON3GQ8MQ/GZM=";
+  };
+
+  buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/uuid6/default.nix:19]:
```suggestion
  nativeBuildInputs = [
```

File: pkgs/development/python-modules/textstat/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-YLY8+JSfRbuztCBeRBG7wc1m30wIrvElRYEcfm4k8BE=";
+  };
+
+  propagatedBuildInputs = [
+    setuptools
+    pyphen
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/textstat/default.nix:24]:
```suggestion

  checkPhase = ''
    ./test.py
  '';

```

File: pkgs/development/python-modules/textstat/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  disabled = pythonOlder "3.6";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-YLY8+JSfRbuztCBeRBG7wc1m30wIrvElRYEcfm4k8BE=";
+  };
+
+  propagatedBuildInputs = [
+    setuptools
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/textstat/default.nix:21]:
```suggestion

  nativeBuildInputs = [
    setuptools
  ];

  propagatedBuildInputs = [
```

File: pkgs/development/python-modules/python-lorem/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  pname = "python-lorem";
+  disabled = pythonOlder "3.5";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-aokLCuQq6iHpC90MLCcIQxeEArPyx1o6RU1224xZdxY=";
+  };
+
+  buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-lorem/default.nix:19]:
```suggestion
  nativeBuildInputs = [
```

File: pkgs/development/python-modules/python-lorem/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    setuptools
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-lorem/default.nix:22]: Please execute the tests https://github.com/JarryShaw/lorem/blob/master/test_lorem.py

File: pkgs/development/python-modules/python-lorem/default.nix [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, setuptools
+}:
+
+buildPythonPackage rec {
+  version = "1.3.0.post1";
+  pname = "python-lorem";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-lorem/default.nix:10]:
```suggestion
  pname = "python-lorem";
  version = "1.3.0.post1";
```

File: pkgs/development/python-modules/daltonlens/default.nix [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  pname = "daltonlens";
+  disabled = pythonOlder "3.7";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-T7fXlRdFtcVw5WURPqZhCmulUi1ZnCfCXgcLtTHeNas=";
+  };
+
+  buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/daltonlens/default.nix:22]:
```suggestion
  nativeBuildInputs = [
```

File: pkgs/development/python-modules/daltonlens/default.nix [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+, pythonOlder
+, setuptools
+, setuptools-git
+, numpy
+, pillow
+}:
+
+buildPythonPackage rec {
+  version = "0.1.5";
+  pname = "daltonlens";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/daltonlens/default.nix:13]:
```suggestion
  pname = "daltonlens";
  version = "0.1.5";
```

File: pkgs/development/python-modules/daltonlens/default.nix [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  buildInputs = [
+    setuptools
+    setuptools-git
+  ];
+
+  propagatedBuildInputs = [
+    numpy
+    pillow
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/daltonlens/default.nix:31]: Please execute the tests https://github.com/DaltonLens/DaltonLens-Python/tree/master/tests

File: pkgs/by-name/de/devtoolbox/package.nix [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+    license = with licenses; [
+      gpl3Plus
+      cc0
+      lgpl3Only
+      mit
+      unlicense
+    ];
+    mainProgram = "devtoolbox";
+    maintainers = with maintainers; [ aleksana ];
+    platforms = platforms.linux; # webkitgtk_6_0 is broken on darwin
```
Comment (SuperSandro2000) [inline pkgs/by-name/de/devtoolbox/package.nix:116]:
```suggestion
```
but that is alread marked broken, is it? Then that propagates.

File: pkgs/by-name/de/devtoolbox/package.nix [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+, gtksourceview5
+, webkitgtk_6_0
+, gcr_4
+, gdk-pixbuf
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "devtoolbox";
+  version = "1.1.1";
+  pyproject = false;
```
Comment (SuperSandro2000) [inline pkgs/by-name/de/devtoolbox/package.nix:21]:
```suggestion
  pyproject = false; # uses meson
```

File: pkgs/by-name/de/devtoolbox/package.nix [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+  src = fetchFromGitHub {
+    owner = "aleiepure";
+    repo = "devtoolbox";
+    rev = "v${version}";
+    hash = "sha256-QFGEA+VhhRlvcch2AJrEzvRJGCSqtvZdMXWUvdAGkoU=";
+  };
+
+  # Upstream is using an incompatible version of ruamel.yaml
+  # May be removed in next update
+  patches = [ ./incompatible_ruamel_yaml.patch ];
```
Comment (SuperSandro2000) [inline pkgs/by-name/de/devtoolbox/package.nix:32]: Is there a commit we can fetch?

**Pull request closed without merging**